src/networking/systems.rs

```diff
@@ -0,0 +1,240 @@
+    pub send_event_reader: EventReader<SendEvent>,
+}
+
+// ECS resources containing a map of active network connections
+#[derive(Default)]
+pub struct NetworkConnections {
+    pub connections: HashMap<ConnectionId, Entity>
+}
+
+// Consume events from ECS and publish them to the correct MPSC to send it over the network
+pub fn receive_net_events(
+    mut commands: Commands,
+    mut session: ResMut<SessionEventListenerState>,
+    mut entities: ResMut<NetworkConnections>,
+    mut net_events: ResMut<Events<ReceiveEvent>>
+) {
+    // Break up `session` in a way that Rust is happy with
+    let session: &mut SessionEventListenerState = &mut session;
+    let SessionEventListenerState { event_receiver, .. } = session;
+
+    // Pull network events from MPSC and publish them
+    while let Ok(event) = event_receiver.try_recv() {
+        match event {
+            // A new connection has opened, allocate an entry in the hashmap
+            ReceiveEvent::Connected(id) => {
+
+                // Create an entity representing this connection
+                commands.spawn((
+                    Connection { id },
+                ));
+                entities.connections.insert(id.clone(), commands.current_entity().expect("`spawn` did not create an entity"));
+
+                // Create a hashmap to hold stream MPSCs for this connection
+                session
+                    .stream_senders
+                    .insert(id.clone(), Default::default());
+            }
+
+            ReceiveEvent::Disconnected(id) => {
+
+                // Delete the entity representing this connection
+                if let Some(e) = entities.connections.remove(&id) {
+                    commands.despawn(e);
+                }
+
+                // drop all stream senders
+                session.stream_senders.remove(&id);
+            }
+
+            // Store the MPSC to send to this stream in the session state
+            ReceiveEvent::OpenedStream {
+                connection_id,
+                stream_id,
+                ref sender,
+            } => {
+                session
+                    .stream_senders
+                    .entry(connection_id.clone())
+                    .or_default()
+                    .insert(stream_id.clone(), sender.clone());
+            }
+
+            // When the socket closes throw away all session state
+            ReceiveEvent::SocketClosed => {
+
+                // Delete all connection entities
+                for (_, e) in entities.connections.drain() {
+                    commands.despawn(e);
+                }
+
+                // drop all stream senders
+                session.stream_senders.clear();
+            }
+
+            _ => {}
+        }
+
+        // Publish event for other systems to consume
+        net_events.send(event);
+    }
+}
+
+// Take ECS events and forward them to MPSCs to be sent over the network
```

Comment:
```suggestion
/// Take ECS events and forward them to MPSCs to be sent over the network
```
Might as well turn this into a doc comment since you already did the work to explain it